### PR TITLE
ENG-8392: Change repo local account resource to support Kubernetes Secret

### PR DESCRIPTION
**Pull request by VictorGFM**
## Description of the change

ENG-8392: Terraform provider - add resource for k8s secrets local account

- Add support for Kubernetes Secret
- Deprecate `enviroment_variable` field since it contains a typo
- Fix issue with `repository_id` by adding ForceNew to this field
- Fix some details in docs.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] Jira issue referenced in commit message and/or PR title

### Testing

Manual and automated tests

### CODE REVIEW

**Comment by VictorGFM** [comment]
@wcmjunior , while I was working on these changes I noticed some limitations regarding the way the provider is structured/organized, more specifically about the way the [resource.go](https://github.com/cyralinc/terraform-provider-cyral/blob/main/cyral/resource.go) is currently designed. This file defines a [SchemaDataHandler](https://github.com/cyralinc/terraform-provider-cyral/blob/main/cyral/resource.go#L16) interface with two methods: [WriteToSchema](https://github.com/cyralinc/terraform-provider-cyral/blob/main/cyral/resource.go#L17) and [ReadFromSchema](https://github.com/cyralinc/terraform-provider-cyral/blob/main/cyral/resource.go#L18), they are used to manage the CRUD operations for almost all the resources. What I noticed is that currently if an error occurs within these methods, is not possible to return and handle these errors which limit the way we can manage the resource lifecycles. In this case, I intended to add a [Config](https://github.com/cyralinc/repo-auth-client-go/blob/main/messages/repo_auth.proto#L269) field to the [repository_local_account](https://github.com/cyralinc/terraform-provider-cyral/blob/main/cyral/resource_cyral_repository_local_account.go) resource, since this field is present in our API but is missing in the resource, but because of this limitation I mentioned, there was not a good way to add the `Config` field to the `repository_local_account` (since it involved handling a parse operation that could throw an error) and I also ended up spending a considerable time figuring a way to solve that.

That being said, I also believe that there are other problems with the `resource.go` regarding readability and other aspects, because of that , even though this is not something of high priority, I think we should consider refactoring this file, which would require us to create a specific task for that, since this file is used by almost all the resources we have. Let me know what you think.

File: cyral/resource_cyral_repository_local_account.go [hunk 1]
```diff
@@ -121,31 +121,86 @@ func (resource *HashicorpVaultResource) ReadFromSchema(d *schema.ResourceData) {
...
+		})
+	} else {
+		d.Set("environment_variable", []interface{}{
+			map[string]interface{}{
+				"database_name": resource.DatabaseName,
+				"local_account": resource.RepoAccount,
+				"variable_name": resource.VariableName,
+			},
+		})
+	}
```
Comment (wcmjunior) [inline cyral/resource_cyral_repository_local_account.go:149]: Had a hard time finding the actual difference and saw that it was the typo in `environment`. I think the best make this code easier to follow is to just have in the conditional the two strings with and without the typo as follows:
```suggestion
	argument := "environment_variable"
	if useDeprecatedArgument {
		// Deprecated: should be removed in the next MAJOR release
		argument = "enviroment_variable"
	}
	d.Set(argument, []interface{}{
		map[string]interface{}{
			"database_name": resource.DatabaseName,
			"local_account": resource.RepoAccount,
			"variable_name": resource.VariableName,
		},
	})
```

File: cyral/resource_cyral_repository_local_account.go [hunk 2]
```diff
@@ -269,15 +322,11 @@ func resourceRepositoryLocalAccount() *schema.Resource {
...
-		ConflictsWith: []string{
-			"aws_iam",
-			"cyral_storage",
-			"hashicorp_vault",
-			"enviroment_variable",
-		},
+	awsSecretsManagerSchema := &schema.Schema{
+		Type:         schema.TypeSet,
+		Optional:     true,
+		ExactlyOneOf: secretManagersTypes,
```
Comment (wcmjunior) [inline cyral/resource_cyral_repository_local_account.go:328]: Much better than using the `ConflictsWith`. :-)

File: cyral/resource_cyral_repository_local_account.go [hunk 3]
```diff
@@ -378,14 +440,44 @@ func resourceRepositoryLocalAccount() *schema.Resource {
 		},
 	}
 
+	kubernetesSecretSchema := &schema.Schema{
```
Comment (wcmjunior) [inline cyral/resource_cyral_repository_local_account.go:443]:
```suggestion
	kubernetesSecretSchema := &schema.Schema{
		Description: "",
```

File: cyral/resource_cyral_repository_local_account.go [hunk 4]
```diff
@@ -378,14 +440,44 @@ func resourceRepositoryLocalAccount() *schema.Resource {
...
+		Type:         schema.TypeSet,
+		Optional:     true,
+		ExactlyOneOf: secretManagersTypes,
+		MaxItems:     1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"database_name": {
```
Comment (wcmjunior) [inline cyral/resource_cyral_repository_local_account.go:450]:
```suggestion
				"database_name": {
					Description: "",
```

**Review by wcmjunior — COMMENTED** [review]
Please add descriptions to the schema as in [this PR](https://github.com/cyralinc/terraform-provider-cyral/pull/183).

File: cyral/resource_cyral_repository_local_account.go (hunk 1 above)
Comment (VictorGFM) [reply cyral/resource_cyral_repository_local_account.go:149]: That's way better, thanks for the suggestion!

**Review by wcmjunior — APPROVED** [review]

**Review by ccampo133 — APPROVED** [review]

**Comment by VictorGFM** [comment]
> Sorry, @VictorGFM , I did not see this message. TBH, we didn't spend much though on this `resource.go`. The goal back then was just to eliminate code replication as we had several resources that would look mostly the same. I'm totally in favor of refactoring it, but we should spend more time discussing the pros and cons of any new approach. Please create an issue in this repo so we can let this one for public discussion. Thanks!

Sure, I'll create an issue for that.

**Pull request merged**